crates/runix/src/flake_ref/indirect.rs

```diff
@@ -65,6 +71,17 @@ impl IndirectRef {
             attributes,
         }
     }
+
+    /// Resolves an indirect flake reference to a concrete reference
+    ///
+    /// Note that this method calls `parser-util`, which relies on the `NIX_USER_CONF_FILES`
+    /// environment variable to be set and contain conf files that point to custom registries
+    /// that you want to use for resolution, otherwise only the user's local registry is used.
+    pub fn resolve(&self) -> Result<FlakeRef, UrlParseError> {
+        let json = serde_json::to_string(&self.attributes)?;
```

Comment:
We need to serialize the type bit as well, `self.attributes` only encodes the additional attributes
```suggestion
        let json = serde_json::to_string(&self)?;
```

Reply:
note: fixed in #48